lib/devmapper_mock.ml

```diff
@@ -324,68 +197,101 @@ module DeviceSet = struct
-  DeviceSet.save_file !devices !persistent_fn
+  let reload device targets =
+    DeviceSet.reload !devices device targets;
+    DeviceSet.save_file !devices !persistent_fn
 
-let reload device targets =
-  DeviceSet.reload !devices device targets;
-  DeviceSet.save_file !devices !persistent_fn
+  let suspend device =
+    DeviceSet.suspend !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let suspend device =
-  DeviceSet.suspend !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let resume device =
+    DeviceSet.resume !devices device;
+    DeviceSet.save_file !devices !persistent_fn
 
-let resume device =
-  DeviceSet.resume !devices device;
-  DeviceSet.save_file !devices !persistent_fn
+  let mknod device path mode =
+    DeviceSet.mknod !devices device path mode;
+    DeviceSet.save_file !devices !persistent_fn
 
-let mknod device path mode =
-  DeviceSet.mknod !devices device path mode;
-  DeviceSet.save_file !devices !persistent_fn
+  let stat device =
+    let ret = DeviceSet.stat !devices device in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let stat device =
-  let ret = DeviceSet.stat !devices device in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let ls () =
+    let ret = DeviceSet.ls !devices () in
+    DeviceSet.save_file !devices !persistent_fn;
+    ret
 
-let ls () =
-  let ret = DeviceSet.ls !devices () in
-  DeviceSet.save_file !devices !persistent_fn;
-  ret
+  let clear () =
+    DeviceSet.clear !devices ();
+    DeviceSet.save_file !devices !persistent_fn
 
-let clear () =
-  DeviceSet.clear !devices ();
-  DeviceSet.save_file !devices !persistent_fn
+  let get_persistent () =
+    !persistent_fn
 
-let get_persistent () =
-  !persistent_fn
+  let set_persistent filename =
+    persistent_fn := filename
 
-let set_persistent filename =
-  persistent_fn := filename
+  let save_file path =
+    DeviceSet.save_file !devices path
 
-let save_file path =
-  DeviceSet.save_file !devices path
+  let load_file path =
+    devices := DeviceSet.load_file path
+end
+
+let create = Mock.create
+let remove = Mock.remove
+let reload = Mock.reload
+let suspend = Mock.suspend
+let resume = Mock.resume
+let mknod = Mock.mknod
+let stat = Mock.stat
+let ls = Mock.ls
+let clear = Mock.clear
+let get_persistent = Mock.get_persistent
+let set_persistent = Mock.set_persistent
+let save_file = Mock.save_file
+let load_file = Mock.load_file
+
+module type DM_intf = sig
+  val create: device -> Devmapper.Target.t list -> unit
+  val remove: device -> unit
+  val reload: device -> Devmapper.Target.t list -> unit
+  val suspend: device -> unit
+  val resume: device -> unit
+  val mknod: device -> string -> int -> unit
+  val stat: device -> Devmapper.info option
+  val ls: unit -> device list
+end
 
-let load_file path =
-  devices := DeviceSet.load_file path
+type dm_intf_type = DMReal | DMMock
 
+let select_dm (intf_type : dm_intf_type) : (module DM_intf) =
```

Comment:
I think this function could be moved into xenvm itself. Xenvm can parse the command-line arguments and decide which version to use.